sql: apply zone configs for copied INDEX for ALTER PK on RBR tables

otan opened the pull request:
Release justification: bug fix for existing feature

Release note (bug fix): Previously, when using ALTER PRIMARY KEY on a
REGIONAL BY ROW table, the copied unique index from the old PRIMARY KEY
would not have the correct zone configurations applied. This commit
fixes that. Users who encountered this bug should re-create the
index.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69681)
<!-- Reviewable:end -->

otan committed:
sql: apply zone configs for copied INDEX for ALTER PK on RBR tables

Release justification: bug fix for existing feature

Release note (bug fix): Previously, when using ALTER PRIMARY KEY on a
REGIONAL BY ROW table, the copied unique index from the old PRIMARY KEY
would not have the correct zone configurations applied. This commit
fixes that. Users who encountered this bug should re-create the
index.

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69681#-:-MiWGIbd5oQ-XDbLovf5:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajwerner
